### PR TITLE
Stream selector should not look up models twice

### PR DESCRIPTION
**Pull request by rynbyjn**
[Finishes #139985727](https://www.pivotaltracker.com/story/show/139985727)

### CODE REVIEW

**Comment by mkitt** [comment]
@rynbyjn Did Danger fail this on something?

File: src/selectors/stream.js [hunk 1]
```diff
@@ -17,53 +18,26 @@ export const selectStreamMappingType = state => state.stream.getIn(['meta', 'map
...
-  return meta.resultKey || selectPathname(state)
-}
-
-const selectStreamDeletions = (state, props) => {
-  const meta = selectMeta(state, props)
-  const resultPath = meta.resultKey || selectPathname(state)
-  const result = state.json.getIn(['pages', resultPath], Immutable.Map({ ids: [] }))
-  return (result.get('type') === meta.mappingType) ||
-    (meta.resultFilter && result.get('type') !== meta.mappingType)
-}
```
Comment (rynbyjn) [inline src/selectors/stream.js]: I think we created these before we knew about `createSelector` or something. Also, it didn't seem like we really needed `selectStreamDeletions` and I couldn't find where the logic would really matter.

File: .travis.yml [hunk 2]
```diff
@@ -5,7 +5,7 @@ script:
 - yarn test
 - rvm install 2.3.1
 - rvm use 2.3.1
-- gem install danger
+- gem install danger -v 2.0.1
```
Comment (rynbyjn) [inline .travis.yml]: Newer versions of danger don't run the organization's Dangerfile before running a particular project's Dangerfile. They now want you to [make a gem](http://danger.systems/guides/faq.html#i-want-to-run-danger-across-multiple-repos) that gets included in a project's Dangerfile, but that sounded like a pain, so going to try to run what the mothership uses.

File: src/selectors/stream.js (hunk 1 above)
Comment (mkitt) [inline src/selectors/stream.js]: 👍

File: src/containers/StreamContainer.js [hunk 3]
```diff
@@ -33,22 +33,19 @@ const selectActionPath = props =>
...
       isGridMode: selectIsGridMode(state),
       omnibar: selectOmnibar(state),
-      result,
-      resultPath,
+      result: selectStreamFilteredResult(state, props),
+      resultPath: selectStreamResultPath(state, props),
       stream: selectStream(state),
-    }
-  }
+    })
```
Comment (mkitt) [inline src/containers/StreamContainer.js]: Nice cleanup on this one.

**Pull request merged**